src/loaders/workerTypes.js

```diff
@@ -49,11 +50,41 @@ export default ({ queue, awsProvisioner }) => {
       })
     )
   );
+  const awsProvisionerRecentErrors = new DataLoader(queries =>
+    Promise.all(
+      queries.map(async ({ filter }) => {
+        const recentErrors = (await ec2Manager.getRecentErrors()).errors;
+        const { region, az, instanceType, ...errors } = filter
+          ? sift(filter, recentErrors)
+          : recentErrors;
+
+        return new Aws(region, az, instanceType, errors);
+      })
+    )
+  );
+  const awsProvisionerHealth = new DataLoader(queries =>
+    Promise.all(
+      queries.map(async ({ filter }) => {
+        const data = await ec2Manager.getHealth();
+        const health = filter ? sift(filter, data) : data;
+        const createEntity = ({ region, az, instanceType, ...health }) =>
+          new Aws(region, az, instanceType, health);
+
+        return {
+          running: health.running.map(createEntity),
+          terminationHealth: health.terminationHealth.map(createEntity),
+          requestHealth: health.requestHealth.map(createEntity),
+        };
```

Comment:
Having this as an entity will require an extra 3 `map`s. Not sure if this is more efficient, but I can see how it's more readable.